cmd/tar2ext4/tar2ext4.go

```diff
@@ -1,63 +1,317 @@
+	gptFlag            = "gpt"
+
+	// common flags
+	inputFlag = "input"
+
+	// flags for reading gpt disks
+	pmbrFlag               = "pmbr"
+	headerFlag             = "header"
+	altHeaderFlag          = "alt-header"
+	entryFlag              = "entry"
+	altEntryFlag           = "alt-entry"
+	readAllBytesFlag       = "all"
+	partitionFlag          = "partition"
+	paritionSuperblockFlag = "partition-sb"
+	vhdFooterFlag          = "vhd-footer"
 )
 
 func main() {
-	flag.Parse()
-	if flag.NArg() != 0 || len(*output) == 0 {
-		flag.Usage()
-		os.Exit(1)
+	app := cli.NewApp()
+	app.Name = "tar2ext4"
+	app.Usage = "converts tar file(s) into vhd(s)"
+	app.Commands = []cli.Command{
+		readGPTCommand,
 	}
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  outputFlag + ",o",
+			Usage: "output file",
+		},
+		cli.BoolFlag{
+			Name:  overlayFlag,
+			Usage: "produce overlayfs-compatible layer image",
+		},
+		cli.BoolFlag{
+			Name:  vhdFlag,
+			Usage: "add a VHD footer to the end of the image",
+		},
+		cli.BoolFlag{
+			Name:  inlineFlag,
+			Usage: "write small file data into the inode; not compatible with DAX",
+		},
+		cli.StringSliceFlag{
+			Name:  inputFlag + ",i",
+			Usage: "input file(s)",
+		},
+		cli.StringSliceFlag{
+			Name:  partitionGUIDsFlag + ",g",
+			Usage: "guid(s) to use for gpt partitions",
+		},
+		cli.StringFlag{
+			Name:  diskGUIDFlag,
+			Usage: "guid for the resulting gpt disk",
+		},
+		cli.BoolFlag{
+			Name:  gptFlag,
+			Usage: "indicates if this disk should be produced as a gpt disk",
+		},
+	}
+	app.Action = func(cliCtx *cli.Context) (err error) {
+		defer func() {
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+			}
+		}()
+		fileNames := cliCtx.StringSlice(inputFlag)
+		inputs := []io.Reader{}
+		if len(fileNames) == 0 {
+			inputs = append(inputs, os.Stdin)
+		}
 
-	err := func() (err error) {
-		in := os.Stdin
-		if *input != "" {
-			in, err = os.Open(*input)
+		for _, p := range fileNames {
+			in, err := os.Open(p)
 			if err != nil {
 				return err
 			}
+			inputs = append(inputs, in)
 		}
-		out, err := os.Create(*output)
+		outputName := cliCtx.String(outputFlag)
+		out, err := os.Create(outputName)
 		if err != nil {
 			return err
 		}
 
 		var opts []tar2ext4.Option
-		if *overlay {
+		if cliCtx.Bool(overlayFlag) {
 			opts = append(opts, tar2ext4.ConvertWhiteout)
 		}
-		if *vhd {
+		if cliCtx.Bool(vhdFlag) {
 			opts = append(opts, tar2ext4.AppendVhdFooter)
 		}
-		if *inlineData {
+		if cliCtx.Bool(inlineFlag) {
 			opts = append(opts, tar2ext4.InlineData)
 		}
-		err = tar2ext4.Convert(in, out, opts...)
-		if err != nil {
-			return err
+
+		if cliCtx.Bool(gptFlag) {
+			diskGUID := cliCtx.String(diskGUIDFlag)
+			if diskGUID == "" {
+				return fmt.Errorf("must provide a disk guid when creating gpt disk")
+			}
+			partitionGUIDs := cliCtx.StringSlice(partitionGUIDsFlag)
+			if len(partitionGUIDs) == 0 {
+				return fmt.Errorf("must provide partition guids when creating gpt disk")
```

Comment:
minor nit: `errors.New` instead of `fmt.Errorf`, since theres no formatting directives